### PR TITLE
[SYCL] Adjust Driver and tests for addition of imf-bf16 library.

### PR DESCRIPTION
**Pull request by rdeodhar**
This change adds the lib-imf-bf16 library into the libraries expected to be linked in by the Driver in various scenarios.

### CODE REVIEW

File: clang/test/Driver/sycl-bfloat16-lib.cpp [hunk 1]
```diff
@@ -16,57 +15,59 @@
...
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device pvc" %s -### 2>&1 \
-// RUN:   | FileCheck %s -check-prefix=BFLOAT16-FALLBACK-NATIVE
+// R UN: %clangxx -fsycl -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device pvc" %s -### 2>&1 \
+// R UN:   | FileCheck %s -check-prefix=BFLOAT16-FALLBACK-NATIVE
 
 // test that an AOT-CPU + AOT-Gen9 compilation uses fallback + native libs
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" %s -### 2>&1 \
-// RUN:   | FileCheck %s -check-prefix=BFLOAT16-FALLBACK-FALLBACK
+// R UN: %clangxx -fsycl -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device gen9" %s -### 2>&1 \
+// R UN:   | FileCheck %s -check-prefix=BFLOAT16-FALLBACK-FALLBACK
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-bfloat16-lib.cpp:51]: What happened here?  Looks like an unintentional space was added to the `RUN:`

Comment (rdeodhar) [reply clang/test/Driver/sycl-bfloat16-lib.cpp:51]: Somehow the full test passes locally but fails in CI.
I am trying to temporarily disable some tests to get to the root cause of the failure.

**Comment by rdeodhar** [comment]
The one fail on AMD/HIP appears to be caused by use of the fp16 data type without checking that the GPU supports it.
There is no use of bfloat16 in the test.

File: clang/test/Driver/sycl-bfloat16-lib-win.cpp [hunk 2]
```diff
@@ -3,52 +3,68 @@
...
 
 /// ###########################################################################
-/// test that no bfloat16 libraries are added in JIT mode
+/// Test that no bfloat16 libraries are added in JIT mode.
 // RUN: %clangxx -fsycl %s --sysroot=%S/Inputs/SYCL -### 2>&1 \
 // RUN:   | FileCheck %s -check-prefix=BFLOAT16
 
-// test that fallback bfloat16 libraries are added in JIT mode with generic target
-// RUN: %clangxx -fsycl -fsycl-targets=spir64 %s -### 2>&1 \
+// Test that fallback bfloat16 libraries are added in JIT mode with \
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-bfloat16-lib-win.cpp:13]: The escape at the end of comment lines is not necessary

Comment (rdeodhar) [reply clang/test/Driver/sycl-bfloat16-lib-win.cpp:13]: Right. I'll remove them.

**Review by mdtoguchi — APPROVED** [review]
LGTM, thanks!

**Review by yubingex007-a11y — APPROVED** [review]

**Comment by yubingex007-a11y** [comment]
why we haven't  merged this PR?

**Comment by MrSidims** [comment]
I believe only @intel/llvm-gatekeepers can merge PRs.

**Comment by MrSidims** [comment]
I'm not 100% sure, but it seems like this PR brings the following regression:
Failed Tests (1):
  SYCL :: BFloat16/bfloat16_builtins.cpp

**Comment by steffenlarsen** [comment]
> why we haven't merged this PR? @mdtoguchi can you merge it on behalf of Rajiv?

Changes have not yet been merged due to the failure. In accordance with the [contribution guidelines](https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#review-and-acceptance-testing) unrelated failures should be mentioned and reasoning should be given for why it is unrelated, optimally with a link to a related issue and/or a PR addressing the unrelated failures.

**Comment by steffenlarsen** [comment]
An issue has been opened for the failure: https://github.com/intel/llvm/issues/8040

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers  guess we can merge this PR. The mentioned issue is known and reported and lucky guess with 'verify with' command makes the test to pass.

**Pull request merged**